app/jobs/bsnl_sms_status_job.rb

```diff
@@ -0,0 +1,50 @@
+class BsnlSmsStatusJob
+  include Sidekiq::Worker
+  include Sidekiq::Throttled::Worker
+  BSNL_TIME_ZONE = "Asia/Kolkata"
+
+  sidekiq_options queue: :default
+  sidekiq_options retry: 3
+
+  sidekiq_throttle(
+    threshold: {limit: 250, period: 1.minute}
```

Comment:
API limit?

Reply:
Not a limitation on the API; We are limiting ourselves in order to not DOS their services, since we can have 15-20k messages on a given day.